nemo/collections/asr/models/ctc_models.py

```diff
@@ -221,6 +224,29 @@ def _setup_dataloader_from_config(self, config: Optional[Dict]):
             augmentor = None
 
         shuffle = config['shuffle']
+        device = 'gpu' if torch.cuda.is_available() else 'cpu'
+        if config.get('use_dali', False):
+            device_id = self.local_rank if device == 'gpu' else -1
```

Comment:
```suggestion
            device_id = self.local_rank if device == 'gpu' else None
```
None stands for GPU only.